lib/pbench/test/unit/server/test_datasets_inventory.py

```diff
@@ -40,21 +40,24 @@ def query_api(
 
         return query_api
 
-    class Tarball:
+    class MockTarball:
         def __init__(self, path: Path, controller: Controller):
             self.name = "dir_name.tar.xz"
             self.tarball_path = path
             self.unpacked = None
 
-        def extract(self, target):
-            info = {"name": "f1.json", "type": CacheType.FILE}
-            stream = "file_as_a_byte_stream"
-            return (info, stream)
+        def filestream(self, target):
+            info = {
+                "name": "f1.json",
+                "type": CacheType.FILE,
+                "stream": "file_as_a_byte_stream",
```

Comment:
`info["stream"]` here is a `str`, not a "stream".  Perhaps this would work instead:
```python
                "stream": io.BytesIO(b"file_as_a_byte_stream"),
```